var/spack/repos/builtin/packages/sundials/package.py

```diff
@@ -172,21 +172,22 @@ class Sundials(CMakePackage, CudaPackage, ROCmPackage):
     depends_on('raja+rocm', when='+raja +rocm')
 
     # External libraries
-    depends_on('lapack',              when='+lapack')
-    depends_on('suite-sparse',        when='+klu')
-    depends_on('petsc+mpi',           when='+petsc')
-    depends_on('hypre+mpi',           when='+hypre')
-    depends_on('superlu-dist@6.1.1:', when='@:5.4.0 +superlu-dist')
-    depends_on('superlu-dist@6.3.0:', when='@5.5.0: +superlu-dist')
-    depends_on('trilinos+tpetra',     when='+trilinos')
+    depends_on('lapack',                  when='+lapack')
+    depends_on('suite-sparse',            when='+klu')
+    depends_on('petsc+mpi',               when='+petsc')
+    depends_on('hypre+mpi~int64',         when='@5.7.1: +hypre ~int64')
+    depends_on('hypre+mpi+int64',         when='@5.7.1: +hypre +int64')
+    depends_on('hypre@:2.22.0+mpi~int64', when='@:5.7.0 +hypre ~int64')
+    depends_on('hypre@:2.22.0+mpi+int64', when='@:5.7.0 +hypre +int64')
```

Comment:
It looks like you merged 

```
depends_on('hypre~int64', when='+hypre ~int64')
depends_on('hypre+int64', when='+hypre +int64')
```

with these constraints. Was that on account of concretization errors?

Reply:
No, there was not, I followed advice from @balos1 to simplify the rules since Hypre dependency has to be with +mpi in any case. Hope I understood the question !

Reply:
I suggested that these were merged as I have been wanting to cleanup the `depends_on` in the sundials package.

Reply:
That is interesting: generally I would say the constraints are clearer when they are not merged. When merging, the number of constraints multiplies: two boolean flags leads to 4 `depends_on` specifications. This is manageable at this level, but would likely become tiresome for anything more complex. That being said I can merge as-is for now.